### DISCUSSION TITLE
Validate limits

### DISCUSSION THREAD

**Discussion by gutogm** [ytetsuro/sabik#304] [Ideas]
To enhance the quality and maintainability of our codebase, we could implement a configuration file that sets acceptable limits for key metrics. The configuration file would allow us to specify the following parameters:

check: a boolean flag that determines whether to throw an error when the limits are exceeded. This flag is particularly useful for continuous integration and deployment checks

**cyclomaticComplexity**: a number that defines the overall acceptable limit for cyclomatic complexity

**cognitiveComplexity**: a number that defines the overall acceptable limit for cognitive complexity

**halsteadComplexityMeasures**: a number that defines the overall acceptable limit for halstead complexity measures

**lineOfCode**: a number that defines the overall acceptable limit for lines of code within a function or method

**maintainability**: a number that defines the overall acceptable limit for maintainability

By implementing this configuration file, we can ensure that our code adheres to specific quality standards and limit the risk of introducing technical debt. The configuration file can be integrated into our development workflow and used to guide code reviews and feedback. This will ultimately lead to a more consistent and maintainable codebase.

An possible format
```typescript
{
  "check": boolean,
  "cyclomaticComplexity": number,
  "cognitiveComplexity": number,
  "halsteadComplexityMeasures": number,
  "lineOfCode": number,
  "maintainability": number,

}
```

What do you think?

**Comment by ytetsuro** [comment]
Thanks comment.

I think this is a good idea.

However, we also want Sabik to be a helper to analyze and find refactoring targets, not a linter.
If I have time I would like to write an eslint plugin using Sabik!

What do you think about my opinion?

**Reply by gutogm** [reply]
I think it's a good idea because in addition to integrations with IDEs, it would give space for the team itself to decide how far to go with these rules

Is this implementation already underway?

**Reply by ytetsuro** [reply]
It is not done yet.
